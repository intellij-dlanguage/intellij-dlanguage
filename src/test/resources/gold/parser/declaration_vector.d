
__vector(int) i;
__vector(V[N]) j;