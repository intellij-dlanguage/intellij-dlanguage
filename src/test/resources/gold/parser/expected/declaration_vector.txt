D Language File
  DLanguageSpecifiedVariableDeclarationImpl(VARIABLE_DECLARATION)
    DLanguageBasicTypeImpl(BASIC_TYPE)
      DLanguageVectorImpl(VECTOR)
        PsiElement(DlangTokenType.__vector)('__vector')
        PsiElement(DlangTokenType.()('(')
        DLanguageTypeImpl(TYPE)
          DLanguageBasicTypeImpl(BASIC_TYPE)
            DLanguageBuiltinTypeImpl(BUILTIN_TYPE)
              PsiElement(DlangTokenType.int)('int')
        PsiElement(DlangTokenType.))(')')
    IDENTIFIER_INITIALIZER
      PsiElement(DlangTokenType.ID)('i')
    PsiElement(DlangTokenType.;)(';')
  DLanguageSpecifiedVariableDeclarationImpl(VARIABLE_DECLARATION)
    DLanguageBasicTypeImpl(BASIC_TYPE)
      DLanguageVectorImpl(VECTOR)
        PsiElement(DlangTokenType.__vector)('__vector')
        PsiElement(DlangTokenType.()('(')
        DLanguageTypeImpl(TYPE)
          DLanguageBasicTypeImpl(BASIC_TYPE)
            DLanguageQualifiedIdentifierImpl(QUALIFIED_IDENTIFIER)
              PsiElement(DlangTokenType.ID)('V')
              PsiElement(DlangTokenType.[)('[')
              DLanguageReferenceExpressionImpl(REFERENCE_EXPRESSION)
                PsiElement(DlangTokenType.ID)('N')
              PsiElement(DlangTokenType.])(']')
        PsiElement(DlangTokenType.))(')')
    IDENTIFIER_INITIALIZER
      PsiElement(DlangTokenType.ID)('j')
    PsiElement(DlangTokenType.;)(';')